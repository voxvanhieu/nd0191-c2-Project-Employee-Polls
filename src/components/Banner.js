import React from 'react';

export { Banner };

function Banner({ text }) {

    return (
        <div
            className='relative overflow-hidden flex items-center border border-default-200 dark:border-default-100 px-2 py-4 rounded-lg bg-gradient-to-r bg-gradient-to-r from-indigo-500 via-purple-500 to-pink-500 text-center mb-5'
            shadow="sm">
            <h2 data-testid="lblBanner" className="font-semibold text-xl text-foreground/90 w-full text-center">{text}</h2>
        </div>
    );
}