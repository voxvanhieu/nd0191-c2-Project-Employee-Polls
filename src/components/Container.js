import React from 'react';


export { Container };

function Container({ children }) {
    return (
        <div className="container mx-auto min-h-svh mt-8">
            {children}
        </div>
    );
}
