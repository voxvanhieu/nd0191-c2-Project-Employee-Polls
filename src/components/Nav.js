import React from "react";
import { useLocation } from "react-router-dom";
import {
    Navbar,
    NavbarBrand,
    NavbarContent,
    NavbarItem,
    Link,
    Button
} from "@nextui-org/react";

import { history } from "../helpers";
import { NavBarAuthMenu } from "./NavBarAuthMenu";

export { Nav };

function Nav() {

    history.location = useLocation();

    function isActiveThisMenu(menuItem) {
        return menuItem === history.location.pathname
    }

    return (
        <Navbar
            classNames={{
                item: [
                    "flex",
                    "relative",
                    "h-full",
                    "items-center",
                    "data-[active=true]:after:content-['']",
                    "data-[active=true]:after:absolute",
                    "data-[active=true]:after:bottom-0",
                    "data-[active=true]:after:left-0",
                    "data-[active=true]:after:right-0",
                    "data-[active=true]:after:h-[2px]",
                    "data-[active=true]:after:rounded-[2px]",
                    "data-[active=true]:after:bg-primary",
                ],
            }}>
            <NavbarBrand>
                <p className="font-bold text-inherit">Employee Polls</p>
            </NavbarBrand>
            <NavbarContent className="hidden sm:flex gap-4" justify="center">
                <NavbarItem isActive={isActiveThisMenu("/")}>
                    <Link color="foreground" href="/">
                        Home
                    </Link>
                </NavbarItem>
                <NavbarItem isActive={isActiveThisMenu("/leaderboard")}>
                    <Link color="foreground" href="/leaderboard">
                        Leaderboard
                    </Link>
                </NavbarItem>
                <NavbarItem isActive={isActiveThisMenu("/add")}>
                    <Link color="foreground" href="/add">
                        New
                    </Link>
                </NavbarItem>
                <NavbarItem isActive={isActiveThisMenu("/about")}>
                    <Link color="foreground" href="/about">
                        About
                    </Link>
                </NavbarItem>
            </NavbarContent>
            <NavbarContent justify="end">
                <NavBarAuthMenu />
            </NavbarContent>
        </Navbar >
    );
}
