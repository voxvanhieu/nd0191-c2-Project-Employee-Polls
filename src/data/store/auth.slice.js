import { createSlice, createAsyncThunk, createReducer } from "@reduxjs/toolkit";

import * as API from '../api';
import { alertActions } from "./alert.slice";
import { history } from '../../helpers'

// Create slice
const name = 'auth';
const initialState = createInitialState();
const selectors = createSelectors();
const reducers = createReducers();
const extraActions = createExtraActions();
const slice = createSlice({ name, initialState, reducers, selectors });

// Exports
export const authActions = { ...slice.actions, ...extraActions };
export const authSelectors = { ...slice.selectors }
export const authReducer = slice.reducer;

// Implementation

function createInitialState() {
    return {
        value: JSON.parse(localStorage.getItem('auth'))
    }
}

function createSelectors() {
    return {
        selectValue: (sliceState) => sliceState.value,
    }
}

function createReducers() {

    return {
        setAuth
    }

    function setAuth(state, action) {
        state.value = action.payload;
    }
}

function createExtraActions() {

    return {
        login: _login(),
        logout: _logout()
    }

    function _login() {
        return createAsyncThunk(
            `${name}/login`,
            async ({ username, password }, { dispatch }) => {
                // dispatch(alertActions.clear());
                try {
                    let user = await API.authenticateUser({ username, password });

                    dispatch(authActions.setAuth(user));

                    localStorage.setItem('auth', JSON.stringify(user));

                    if (user && user.name) {
                        dispatch(alertActions.success("Welcome"));
                        setTimeout(() => {
                            const { from } = history.location.state || { from: { pathname: '/' } };
                            history.navigate(from);
                        }, 500);
                    } else {
                        dispatch(alertActions.error("Wrong username or password"));
                    }
                } catch (error) {
                    console.log(error);
                }
            }
        )
    }

    function _logout() {
        return createAsyncThunk(
            `${name}/logout`,
            function (arg, { dispatch }) {
                dispatch(authActions.setAuth(null));
                localStorage.removeItem('auth');
                history.navigate('/login');
            }
        )
    }
}