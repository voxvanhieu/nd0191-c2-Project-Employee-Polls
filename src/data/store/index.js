import { configureStore } from '@reduxjs/toolkit';

import { alertReducer } from './alert.slice';
import { authReducer } from './auth.slice'
import { userReducer as usersReducer } from './user.slice'
import { pollsReducer } from './poll.slice';

export * from './alert.slice'
export * from './auth.slice'
export * from './user.slice'
export * from './poll.slice'

export const store = configureStore({
    reducer: {
        alert: alertReducer,
        auth: authReducer,
        users: usersReducer,
        polls: pollsReducer,
    }
})