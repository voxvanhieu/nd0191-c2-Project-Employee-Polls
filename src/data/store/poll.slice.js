import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";

import * as API from '../api';
import { useDispatch } from "react-redux";

// Create slice
const name = 'polls';
const initialState = createInitialState();
const selectors = createSelectors();
const advSelectors = createAdvancedSelectors();
const reducers = createReducers();
const extraActions = createExtraActions();
const extraReducers = createExtraReducers();

const slice = createSlice({ name, initialState, reducers, extraReducers, selectors });

// Exports
export const pollActions = { ...slice.actions, ...extraActions };
export const pollSelectors = { ...slice.selectors, ...advSelectors }
export const pollsReducer = slice.reducer;

// Implementation

function createInitialState() {
    return {
        allIds: [],
        byId: {},
        status: {}
    }
}

function createSelectors() {
    return {
        selectValue: (sliceState) => sliceState.byId,
    }
}

//  Selector with parameters
function createAdvancedSelectors() {
    return {
        selectQuestionById: (pollId) => (state) => {
            const sliceState = state[slice.name];
            return sliceState.allIds.includes(pollId)
                ? sliceState.byId[pollId]
                : null;
        },
        selectQuestionIds: (pollIds) => (state) => {
            const sliceState = state[slice.name];
            return sliceState.allIds.filter(key => pollIds.includes(key));
        },
        selectQuestionIdsExcepts: (pollIds) => (state) => {
            const sliceState = state[slice.name];
            return sliceState.allIds.filter(key => !pollIds.includes(key));
        }
    }
}


function createReducers() {
    return {
        getById: {
            reducer: (state, action) => state.byId[action.payload.id],
            prepare: (id) => ({ payload: id })
        },
    };
}

function createExtraActions() {

    return {
        getQuestions: createAsyncThunk(
            `${name}/getQuestions`,
            async () => await API._getQuestions()
        ),
    }
}

function createExtraReducers() {
    return (builder) => {
        buildReducerGetQuestions();

        function buildReducerGetQuestions() {
            var { pending, fulfilled, rejected } = extraActions.getQuestions;
            builder
                .addCase(pending, (state) => {
                    state.status = { loading: true };
                })
                .addCase(fulfilled, (state, action) => {
                    state.byId = action.payload;
                    state.allIds = Object.keys(state.byId);
                    state.status = { loading: false };
                })
                .addCase(rejected, (state, action) => {
                    state.status = { error: action.error };
                })
        }
    }
}
