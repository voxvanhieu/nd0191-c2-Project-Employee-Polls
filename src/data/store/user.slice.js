import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";

import * as API from '../api';

// Create slice
const name = 'users';
const initialState = createInitialState();
const selectors = createSelectors();
const advSelectors = createAdvancedSelectors();
const extraActions = createExtraActions();
const extraReducers = createExtraReducers();
const slice = createSlice({ name, initialState, extraReducers, selectors });

// Exports
export const userActions = { ...slice.actions, ...extraActions };
export const userSelectors = { ...slice.selectors, ...advSelectors };
export const userReducer = slice.reducer;

// Implementation

function createInitialState() {
    return {
        allIds: [],
        byId: {},
        status: {}
    }
}

function createSelectors() {
    return {
        selectValue: (sliceState) => sliceState.byId,
    }
}

//  Selector with parameters
function createAdvancedSelectors() {
    return {
        selectUserInfoById: (userId) => (state) => {
            const sliceState = state[slice.name];
            return sliceState.allIds.includes(userId) ? sliceState.byId[userId] : null;
        },
        selectCompletedPolls: (userId) => (state) => {
            const sliceState = state[slice.name];
            return sliceState.allIds.includes(userId) ? sliceState.byId[userId].answers : null;
        }
    }
}

function createExtraActions() {

    return {
        getUsers: createAsyncThunk(
            `${name}/getUsers`,
            async () => await API._getUsers()
        )
    }
}

function createExtraReducers() {
    return (builder) => {
        buildReducerGetUsersAsync();

        function buildReducerGetUsersAsync() {
            var { pending, fulfilled, rejected } = extraActions.getUsers;
            builder
                .addCase(pending, (state) => {
                    state.status = { loading: true };
                })
                .addCase(fulfilled, (state, action) => {
                    state.byId = action.payload;
                    state.allIds = Object.keys(action.payload);
                    state.status = { loading: false };
                })
                .addCase(rejected, (state, action) => {
                    state.status = {
                        loading: false,
                        error: action.error
                    };
                })
        }
    }
}