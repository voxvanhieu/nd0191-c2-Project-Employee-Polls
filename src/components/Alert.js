import React from 'react';
import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useLocation } from 'react-router-dom';

import { alertActions } from '../data/store';
import { Button } from '@nextui-org/react';

export { Alert };

function Alert() {
    const dispatch = useDispatch();
    const location = useLocation();
    const alert = useSelector(x => x.alert.value);

    useEffect(() => {
        // clear alert on location change
        dispatch(alertActions.clear());
    }, [location]);

    if (!alert) return null;

    return (
        <div className="container w-full h-fit bg-warning">
            <div className="m-3 p-3">
                <div className={`flex flex-row justify-between items-center alert alert-dismissible ${alert.type}`}>
                    <p data-testid="lblAlert" className="text-danger">{alert.message}</p>
                    <Button className="btn-close" onClick={() => dispatch(alertActions.clear())}>Close Alert</Button>
                </div>
            </div>
        </div>
    );
}
