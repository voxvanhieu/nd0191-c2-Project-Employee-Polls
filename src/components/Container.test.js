import React from 'react';
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { render, screen } from '@testing-library/react';
import { store } from '../data/store';
import { Alert } from './Alert';
import { Banner } from './Banner';
import { Container } from './Container';
import { Nav } from './Nav';

describe('App snapshot test', () => {
    it('will match snapshot', () => {
        const testText = "This is a sample text for testing";
        let component = render(
            <React.StrictMode>
                <Provider store={store}>
                    <BrowserRouter>
                        <Container>
                            <Alert />
                            <Nav />
                            <Banner text={testText} />
                        </Container>
                    </BrowserRouter>
                </Provider>
            </React.StrictMode>
        );

        expect(component).toBeDefined();
        expect(component).toMatchSnapshot();

        const lblBanner = screen.getByTestId("lblBanner");
        expect(lblBanner.textContent).toBe(testText);
    })
})
