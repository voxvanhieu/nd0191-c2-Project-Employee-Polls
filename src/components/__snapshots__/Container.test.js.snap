// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot test will match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container mx-auto min-h-svh mt-8"
      >
        <nav
          class="flex z-40 w-full h-auto items-center justify-center data-[menu-open=true]:border-none sticky top-0 inset-x-0 backdrop-blur-lg data-[menu-open=true]:backdrop-blur-xl backdrop-saturate-150 bg-background/70"
          style="--navbar-height: 4rem;"
        >
          <header
            class="z-40 flex px-6 gap-4 w-full flex-row relative flex-nowrap items-center justify-between h-[var(--navbar-height)] max-w-[1024px]"
          >
            <div
              class="flex basis-0 flex-row flex-grow flex-nowrap justify-start bg-transparent items-center no-underline text-medium whitespace-nowrap box-border"
            >
              <p
                class="font-bold text-inherit"
              >
                Employee Polls
              </p>
            </div>
            <ul
              class="h-full flex-row flex-nowrap items-center data-[justify=start]:justify-start data-[justify=start]:flex-grow data-[justify=start]:basis-0 data-[justify=center]:justify-center data-[justify=end]:justify-end data-[justify=end]:flex-grow data-[justify=end]:basis-0 hidden sm:flex gap-4"
              data-justify="center"
            >
              <li
                class="text-medium whitespace-nowrap box-border list-none data-[active=true]:font-semibold flex relative h-full items-center data-[active=true]:after:content-[''] data-[active=true]:after:absolute data-[active=true]:after:bottom-0 data-[active=true]:after:left-0 data-[active=true]:after:right-0 data-[active=true]:after:h-[2px] data-[active=true]:after:rounded-[2px] data-[active=true]:after:bg-primary"
                data-active="true"
              >
                <a
                  class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-foreground no-underline hover:opacity-80 active:opacity-disabled transition-opacity"
                  href="/"
                  role="link"
                  tabindex="0"
                >
                  Home
                </a>
              </li>
              <li
                class="text-medium whitespace-nowrap box-border list-none data-[active=true]:font-semibold flex relative h-full items-center data-[active=true]:after:content-[''] data-[active=true]:after:absolute data-[active=true]:after:bottom-0 data-[active=true]:after:left-0 data-[active=true]:after:right-0 data-[active=true]:after:h-[2px] data-[active=true]:after:rounded-[2px] data-[active=true]:after:bg-primary"
              >
                <a
                  class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-foreground no-underline hover:opacity-80 active:opacity-disabled transition-opacity"
                  href="/leaderboard"
                  role="link"
                  tabindex="0"
                >
                  Leaderboard
                </a>
              </li>
              <li
                class="text-medium whitespace-nowrap box-border list-none data-[active=true]:font-semibold flex relative h-full items-center data-[active=true]:after:content-[''] data-[active=true]:after:absolute data-[active=true]:after:bottom-0 data-[active=true]:after:left-0 data-[active=true]:after:right-0 data-[active=true]:after:h-[2px] data-[active=true]:after:rounded-[2px] data-[active=true]:after:bg-primary"
              >
                <a
                  class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-foreground no-underline hover:opacity-80 active:opacity-disabled transition-opacity"
                  href="/add"
                  role="link"
                  tabindex="0"
                >
                  New
                </a>
              </li>
              <li
                class="text-medium whitespace-nowrap box-border list-none data-[active=true]:font-semibold flex relative h-full items-center data-[active=true]:after:content-[''] data-[active=true]:after:absolute data-[active=true]:after:bottom-0 data-[active=true]:after:left-0 data-[active=true]:after:right-0 data-[active=true]:after:h-[2px] data-[active=true]:after:rounded-[2px] data-[active=true]:after:bg-primary"
              >
                <a
                  class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-foreground no-underline hover:opacity-80 active:opacity-disabled transition-opacity"
                  href="/about"
                  role="link"
                  tabindex="0"
                >
                  About
                </a>
              </li>
            </ul>
            <ul
              class="flex gap-4 h-full flex-row flex-nowrap items-center data-[justify=start]:justify-start data-[justify=start]:flex-grow data-[justify=start]:basis-0 data-[justify=center]:justify-center data-[justify=end]:justify-end data-[justify=end]:flex-grow data-[justify=end]:basis-0"
              data-justify="end"
            >
              <li
                class="text-medium whitespace-nowrap box-border list-none data-[active=true]:font-semibold flex relative h-full items-center data-[active=true]:after:content-[''] data-[active=true]:after:absolute data-[active=true]:after:bottom-0 data-[active=true]:after:left-0 data-[active=true]:after:right-0 data-[active=true]:after:h-[2px] data-[active=true]:after:rounded-[2px] data-[active=true]:after:bg-primary"
              >
                <a
                  class="tap-highlight-transparent no-underline hover:opacity-80 active:opacity-disabled transition-opacity z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary/20 text-primary data-[hover=true]:opacity-hover"
                  href="/login"
                  role="button"
                  tabindex="0"
                >
                  Login
                </a>
              </li>
            </ul>
          </header>
        </nav>
        <div
          class="relative overflow-hidden flex items-center border border-default-200 dark:border-default-100 px-2 py-4 rounded-lg bg-gradient-to-r bg-gradient-to-r from-indigo-500 via-purple-500 to-pink-500 text-center mb-5"
          shadow="sm"
        >
          <h2
            class="font-semibold text-xl text-foreground/90 w-full text-center"
            data-testid="lblBanner"
          >
            This is a sample text for testing
          </h2>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container mx-auto min-h-svh mt-8"
    >
      <nav
        class="flex z-40 w-full h-auto items-center justify-center data-[menu-open=true]:border-none sticky top-0 inset-x-0 backdrop-blur-lg data-[menu-open=true]:backdrop-blur-xl backdrop-saturate-150 bg-background/70"
        style="--navbar-height: 4rem;"
      >
        <header
          class="z-40 flex px-6 gap-4 w-full flex-row relative flex-nowrap items-center justify-between h-[var(--navbar-height)] max-w-[1024px]"
        >
          <div
            class="flex basis-0 flex-row flex-grow flex-nowrap justify-start bg-transparent items-center no-underline text-medium whitespace-nowrap box-border"
          >
            <p
              class="font-bold text-inherit"
            >
              Employee Polls
            </p>
          </div>
          <ul
            class="h-full flex-row flex-nowrap items-center data-[justify=start]:justify-start data-[justify=start]:flex-grow data-[justify=start]:basis-0 data-[justify=center]:justify-center data-[justify=end]:justify-end data-[justify=end]:flex-grow data-[justify=end]:basis-0 hidden sm:flex gap-4"
            data-justify="center"
          >
            <li
              class="text-medium whitespace-nowrap box-border list-none data-[active=true]:font-semibold flex relative h-full items-center data-[active=true]:after:content-[''] data-[active=true]:after:absolute data-[active=true]:after:bottom-0 data-[active=true]:after:left-0 data-[active=true]:after:right-0 data-[active=true]:after:h-[2px] data-[active=true]:after:rounded-[2px] data-[active=true]:after:bg-primary"
              data-active="true"
            >
              <a
                class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-foreground no-underline hover:opacity-80 active:opacity-disabled transition-opacity"
                href="/"
                role="link"
                tabindex="0"
              >
                Home
              </a>
            </li>
            <li
              class="text-medium whitespace-nowrap box-border list-none data-[active=true]:font-semibold flex relative h-full items-center data-[active=true]:after:content-[''] data-[active=true]:after:absolute data-[active=true]:after:bottom-0 data-[active=true]:after:left-0 data-[active=true]:after:right-0 data-[active=true]:after:h-[2px] data-[active=true]:after:rounded-[2px] data-[active=true]:after:bg-primary"
            >
              <a
                class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-foreground no-underline hover:opacity-80 active:opacity-disabled transition-opacity"
                href="/leaderboard"
                role="link"
                tabindex="0"
              >
                Leaderboard
              </a>
            </li>
            <li
              class="text-medium whitespace-nowrap box-border list-none data-[active=true]:font-semibold flex relative h-full items-center data-[active=true]:after:content-[''] data-[active=true]:after:absolute data-[active=true]:after:bottom-0 data-[active=true]:after:left-0 data-[active=true]:after:right-0 data-[active=true]:after:h-[2px] data-[active=true]:after:rounded-[2px] data-[active=true]:after:bg-primary"
            >
              <a
                class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-foreground no-underline hover:opacity-80 active:opacity-disabled transition-opacity"
                href="/add"
                role="link"
                tabindex="0"
              >
                New
              </a>
            </li>
            <li
              class="text-medium whitespace-nowrap box-border list-none data-[active=true]:font-semibold flex relative h-full items-center data-[active=true]:after:content-[''] data-[active=true]:after:absolute data-[active=true]:after:bottom-0 data-[active=true]:after:left-0 data-[active=true]:after:right-0 data-[active=true]:after:h-[2px] data-[active=true]:after:rounded-[2px] data-[active=true]:after:bg-primary"
            >
              <a
                class="relative inline-flex items-center tap-highlight-transparent outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 text-medium text-foreground no-underline hover:opacity-80 active:opacity-disabled transition-opacity"
                href="/about"
                role="link"
                tabindex="0"
              >
                About
              </a>
            </li>
          </ul>
          <ul
            class="flex gap-4 h-full flex-row flex-nowrap items-center data-[justify=start]:justify-start data-[justify=start]:flex-grow data-[justify=start]:basis-0 data-[justify=center]:justify-center data-[justify=end]:justify-end data-[justify=end]:flex-grow data-[justify=end]:basis-0"
            data-justify="end"
          >
            <li
              class="text-medium whitespace-nowrap box-border list-none data-[active=true]:font-semibold flex relative h-full items-center data-[active=true]:after:content-[''] data-[active=true]:after:absolute data-[active=true]:after:bottom-0 data-[active=true]:after:left-0 data-[active=true]:after:right-0 data-[active=true]:after:h-[2px] data-[active=true]:after:rounded-[2px] data-[active=true]:after:bg-primary"
            >
              <a
                class="tap-highlight-transparent no-underline hover:opacity-80 active:opacity-disabled transition-opacity z-0 group relative inline-flex items-center justify-center box-border appearance-none select-none whitespace-nowrap font-normal subpixel-antialiased overflow-hidden tap-highlight-transparent data-[pressed=true]:scale-[0.97] outline-none data-[focus-visible=true]:z-10 data-[focus-visible=true]:outline-2 data-[focus-visible=true]:outline-focus data-[focus-visible=true]:outline-offset-2 px-4 min-w-20 h-10 text-small gap-2 rounded-medium [&>svg]:max-w-[theme(spacing.8)] transition-transform-colors-opacity motion-reduce:transition-none bg-primary/20 text-primary data-[hover=true]:opacity-hover"
                href="/login"
                role="button"
                tabindex="0"
              >
                Login
              </a>
            </li>
          </ul>
        </header>
      </nav>
      <div
        class="relative overflow-hidden flex items-center border border-default-200 dark:border-default-100 px-2 py-4 rounded-lg bg-gradient-to-r bg-gradient-to-r from-indigo-500 via-purple-500 to-pink-500 text-center mb-5"
        shadow="sm"
      >
        <h2
          class="font-semibold text-xl text-foreground/90 w-full text-center"
          data-testid="lblBanner"
        >
          This is a sample text for testing
        </h2>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
